src/binary/Builtin.cc

```diff
@@ -632,6 +632,61 @@ extern "C" {
     return INT2FIX(strcoll(RSTRING_PTR(str1), RSTRING_PTR(str2)));
   }
 
+  // reads the value of a hash key
+  // used internally by stftime_wrapper
+  static int
+  tm_hash_value(VALUE hash, const char *key)
+  {
+    return NUM2INT(rb_hash_aref(hash, ID2SYM(rb_intern(key))));
+  }
+  
+  // converts a hash to a tm structure
+  // used internally by stftime_wrapper
+  void
+  hash_to_tm(VALUE hash, struct tm *res)
+  {
```

Comment:
It might be a good idea to clear the tm struct in advance using memset. When you look at the definition of tm you can see that is has a few more members thus tools like valgrind might complain about use of uninitialized data when you pass tm to libc functions. The memset call is common practise for ioctl calls.